extensions/wikia/Email/EmailController.class.php

```diff
@@ -763,4 +762,59 @@ private function trackEmailError( \Exception $e, $result ) {
 			'result' => $result,
 		] );
 	}
+
+	/**
+	 * check user authentication key
+	 * @static
+	 * @access public
+	 * @param array $params
+	 * @returns null
+	 */
+	static public function verifyUserSecretKey( $url, $hash_algorithm = 'sha256' ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)